src/github_runner_image_builder/templates/cloud-init.sh.j2

```diff
@@ -186,14 +186,27 @@ function configure_system_users() {
 
 function execute_script() {
     local script_url="$1"
+    local env_vars="$2"
     if [[ -z "$script_url" ]]; then
         echo "Script URL not provided, skipping."
         return
     fi
+    # Write temp environment variables file, load and delete.
+    TEMP_FILE=$(mktemp)
+    IFS=' ' read -r -a vars <<< "$env_vars"
+    for var in "${vars[@]}"; do
+        echo "$var" >> "$TEMP_FILE"
+    done
+    # Source the temporary file and run the script
+    set -a  # Automatically export all variables
+    source "$TEMP_FILE"
+    set +a  # Stop automatically exporting variables
+
```

Comment:
I think there's a risk of overriding a variable set in here using the script_secrets here. I'd avoid defining these variable inside the cloud-init script and find a way to pass these variables ONLY to the external script probably using `env -i ... ./external.sh`

Reply:
I'm wondering if we should start with a fresh environment using `env -i` - the reason being that some env variables (HOME, shell dir, ...etc) may be assumed to be in place from the user scripts. An example would be snap go package which assumes that the go HOME directory to be set in environment variable before installing any go packages.

Reply:
1. check whether any sensitive values are in the openstack cloud init script
2. export the secrets and make them available
---
The env vars in the cloud-init of openstack does not contain sensitive values
```
[   17.960816] cloud-init[1198]: MEMORY_PRESSURE_WRITE=c29tZSAyMDAwMDAgMjAwMDAwMAA=
[   17.961876] cloud-init[1198]: PWD=/
[   17.962402] cloud-init[1198]: SYSTEMD_EXEC_PID=1192
[   17.963055] cloud-init[1198]: LANG=C.UTF-8
[   17.963634] cloud-init[1198]: MEMORY_PRESSURE_WATCH=/sys/fs/cgroup/system.slice/cloud-final.service/memory.pressure
[   17.964878] cloud-init[1198]: INVOCATION_ID=137b1ce9228c4b07b3b5bfb82e0420ee
[   17.965759] cloud-init[1198]: TERM=vt220
[   17.965824] cloud-init[1198]: USER=root
[   17.967892] cloud-init[1198]: SHLVL=1
[   17.969874] cloud-init[1198]: JOURNAL_STREAM=8:9540
[   17.971821] cloud-init[1198]: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/snap/bin
[   17.973798] cloud-init[1198]: _=/usr/bin/env
```
so i believe it should be okay! :)